pkgs/development/python-modules/netcdf4/default.nix

```
@@ -26,21 +39,30 @@ buildPythonPackage rec {
     hdf5
     curl
     libjpeg
-  ];
+  ] ++ (if mpiSupport then [mpi4py mpi openssh] else []);
+
+  #patches=[./skipDapTest.patch];
+
+  makeFlags = lib.optionals mpiSupport [ "CC=mpicc" "CXX=mpicxx" ];
 
-  checkPhase = ''
-    py.test test/tst_*.py
+  checkPhase= ''
+    cd test
+    rm tst_dap.py # does not fail in the nix-shell  
+    python -m unittest tst_*.py
   '';
 
-  # Tests need fixing.
-  doCheck = false;
 
   # Variables used to configure the build process
   USE_NCCONFIG="0";
-  HDF5_DIR = lib.getDev hdf5;
-  NETCDF4_DIR=netcdf;
-  CURL_DIR=curl.dev;
-  JPEG_DIR=libjpeg.dev;
+  HDF5_DIR= lib.getDev hdf5;
+  NETCDF4_DIR="${netcdf}";
+  CURL_DIR="${curl.dev}";
+  JPEG_DIR="${libjpeg.dev}";
+  
+  passthru = {
+    mpiSupport = mpiSupport;
+    inherit mpi;
```

```suggestion
    inherit mpi mpiSupport;
```
and remove previous line